### DISCUSSION TITLE
What Android level are you running xDrip on?

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2200] [Polls]
Currently, you can run xDrip on any Android phone as long as it is Android 4 or above.
It helps to know how many users still need xDrip to run on Android 4.
Please respond below and choose the Android level you are using.

The question is not if you have an Android 4 device and you may like to use xDrip on it.  
The question is if you are actively using a phone that has only Android 4 on it for running xDrip, and if xDrip stops supporting Android 4, you will be in serious trouble because you have, and you will have, no other phone.

**Comment by ChanceHarrison** [comment]
It would be nice if there was some rudimentary app analytics that just reported the bare minimum, stuff like Android version, device model, etc.. Such information seems like it would be quite valuable for being able to answer questions such as "How many users would be affected by dropping support for Android version \<whatever\>."

Of course, a poll is better than nothing. How can we best spread the word about it?

**Reply by Navid200** [reply]
I suspect this was intentionally never done because we don't want to collect any data from users.  Let me emphasize that's just a speculation on my part.

Even if we add this capability and give the user the option to opt out, we will have no idea how many users opt out.
If we collect the info and not give the user the option to opt out, it's not a good idea.  There will be technical issues as well.  What if the user doesn't want to use data and we upload without their knowledge.  
But, more importantly, collecting data without the consent of the user is a very bad practice.

**Reply by ChanceHarrison** [reply]
I certainly understand and appreciate where you are coming from with wanting to be very respectful of user privacy and autonomy. I would like to think that users could be presented with a screen when first using the app (or the first version that has such analytics), much like the End User License Agreement screen. And that screen can explicitly state what data would be collected, how such data would be used, and required explicit opt-in, along with other best-practices. If all of that is done, then I think we could avoid the worst of the issues concerning data collection. I would like to think that the value gained from having a somewhat-comprehensive understanding of the xDrip user base at a non-identifying level (absolutely no personal information), with things like app version, Android version, device model, etc., would be valuable from a development perspective and might justify the time spent in developing such a feature.

> What if the user doesn't want to use data and we upload without their knowledge.

That screen I mentioned could disclose this fact and suggest that users do not opt-in to to the very basic data collection if they are concerned about data usage.

Of course, xDrip doesn't have a lot of developers, gets a lot of feature requests, and arguably needs to pay off a lot of technical debt, so I wouldn't be surprised if this isn't exactly near the top of anyone's priority list. And maybe a poll works well enough, though I don't think we can honestly say it does. How do people become aware of the poll so they can take it? I can imagine it is a very small subset of users, and likely a subset that skews the data (it's not like the poll is a representative sample).

I would be curious about what you think. If done right, might it be worth considering?

**Reply by Navid200** [reply]
All valid points

Please note that I just help.  I am not the head developer.  I am hardly a developer as I have just learned the little bit I know by working on xDrip.  

The reason I opened this poll was just curiosity.  I had no idea how many would take it.  What we see now can be a sign that not many people are aware of these discussions.

What you say about a poll possibly not being a valid representation of all users is a valid point.
I suggest that the exact same concern will exist if we do exactly what you suggest.  There will be no way to know if those who choose to participate will be randomly distributed.  For all we know those who decide not to participate, or those who cannot participate because they have no reasonable access to the internet, are the same people who cannot afford to buy an Android 5 phone.

The fact is that Android 4 coverage was once removed and there were complaints after.  So, it had to be undone.  
Spending time doing something and having to undo it is not something the head developer is looking forward to.  
I wonder if he can let us do it.  I for one don't mind work that has to be undone because I am learning in the process.

Thanks for the conversation

**Reply by tolot27** [reply]
I don't believe that this poll is representative for Android 4 users. If we plan to stop support for an older Android version, it is IMHO more representative to show a warning/poll just on those devices to get feedback how many and how long these people will continue to use this old Android version. Indeed, the responses will still be biased but it is of much higher interest for those people to respond as for anyone else.
And we can asked for the reason why they wan't to continue to use the older Android version and we can find workarounds (i. e. funding, a backward compatibility fork, etc.).

**Comment by enginemr2** [comment]
Not sure if I am sending this to the correct people about 'what version of Android are you running xDrip On'. Personally I am using Android 6. That seems to be a very stable version since I have heard problems with later versions. 

    On Thursday, July 7, 2022 at 07:30:40 PM PDT, Navid ***@***.***> wrote:  
 
 


All valid points

Please note that I just help. I am not the head developer. I am hardly a developer as I have just learned the little bit I know by working on xDrip.

The reason I opened this poll was just curiosity. I had no idea how many would take it. What we see now can be a sign that not many people are aware of these discussions.

What you say about a poll possibly not being a valid representation of all users is a valid point.
I suggest that the exact same concern will exist if we do exactly what you suggest. There will be no way to know if those who choose to participate will be randomly distributed. For all we know those who decide not to participate or those who cannot participate because they have no reasonable access to the internet are the same people who cannot afford to buy an Android 5 phone.

The fact is that Android 4 coverage was once removed and there were complaints after. So, it had to be undone.
Spending time doing something and having to undo it is not something the head developer is looking forward to.
I wonder if he can let us do it. I for one don't mind work that has to be undone because I am learning in the process.

Thanks for a conversation
